lambda_functions/source/eks-existing-clusters/lambda.py

```diff
@@ -0,0 +1,234 @@
+    client = session.client(
+        service_name='eks',
+        region_name=region_name
+    )
+
+    response = client.describe_cluster(name=cluster_name)
+    cluster_arn = response.get('cluster', {}).get('arn')
+    auth_mode = response.get('cluster', {}).get('accessConfig', {}).get('authenticationMode')
+    public_endpoint = response.get('cluster', {}).get('resourcesVpcConfig', {}).get('endpointPublicAccess')
+
+    return cluster_arn, auth_mode, public_endpoint
+
+def check_fargate(session, region_name, cluster_name):
+    client = session.client(
+        service_name='eks',
+        region_name=region_name
+    )
+
+    try:
+        response = client.list_fargate_profiles(
+            clusterName=cluster_name,
+            maxResults=10
+        )
+        if response['fargateProfileNames'] not in []:
+            logger.info('No fargate profiles found, setting node_type to nodegroup...')
+            node_type = 'nodegroup'
+            return node_type
+        else:
+            node_type = 'fargate'
+            return node_type
+    except botocore.exceptions.ClientError as error:
+        logger.error(error)
+        return None
+
+def start_build(cluster_name, cluster_arn, node_type, account_id, region_name):
+    try:
+        session = boto3.session.Session()
+        client = session.client(
+            service_name='codebuild',
+            region_name=REGION
+        )
+        build = client.start_build(
+            projectName=PROJECT,
+            artifactsOverride={
+                'type': 'S3',
+                'location': f'{BUCKET}',
+                'path': 'BuildResults',
+                'name': f'{cluster_name}-{DATE}',
+                'packaging': 'ZIP'
+            },
+            environmentVariablesOverride=[
+                {
+                    'name': 'CLUSTER',
+                    'value': f'{cluster_name}',
+                    'type': 'PLAINTEXT'
+                },
+                {
+                    'name': 'NODE_TYPE',
+                    'value': f'{node_type}',
+                    'type': 'PLAINTEXT'
+                },
+                {
+                    'name': 'CLUSTER_ARN',
+                    'value': f'{cluster_arn}',
+                    'type': 'PLAINTEXT'
+                },
+                {
+                    'name': 'ACCOUNT_ID',
+                    'value': f'{account_id}',
+                    'type': 'PLAINTEXT'
+                },
+                {
+                    'name': 'REGION',
+                    'value': f'{region_name}',
+                    'type': 'PLAINTEXT'
+                }
+            ]
+        )
+        buildId = build.get('build', {}).get('id')
+        logger.info(f'Started build {PROJECT}, buildId {buildId}')
+    except botocore.exceptions.ClientError as error:
+        logger.error(error)
+
+def cfnresponse_send(event, responseStatus, responseData, physicalResourceId=None, noEcho=False):
```

Comment:
consider using https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-lambda-function-code-cfnresponsemodule.html

Reply:
will consider this for next version